### ISSUE TITLE
`PGS_ShapeBoolean.union(PShape, PShape)` returns shape identical to input argument

### ISSUE THREAD

**Issue by puebloDeLaMuerte** [micycle1/PGS#62]
Hi,

i have an array of many PShapes created from tiling an initial square repeatedly using PGS_Processing.slice(). Within that array i pick two neigbouring shapes for unision via the PGS_ShapeBoolean.union(PShape,PShape) function. This works like a charm most of the time and returns a PShape that combines the areas of the two input shapes. Sometimes it doesen't want to do so and instead just returns a PShape that is identical to the second input argument of union(PShape,PShape).

What's confusing is that i cannot tell the difference between cases where it works and where it doesnt. The input shapes look valid to me in both cases...

 Not shure if this is a bug or if you can point me towards common issues with unision that i should look into...

Thx for a reply and thanks for sharing this awesome library!

**Comment by micycle1** [comment]
Can you certify that the coordinates of the input shapes overlap perfectly? I'm guessing this isn't the case somehow. If so, you could put the shapes through `PGS_Conversion.roundVertexCoords()` first.

Perhaps after having been sliced, the adjacent slices have very slight differences in their coordinates -- a small gap which means they cannot be unioned.

**Comment by puebloDeLaMuerte** [comment]
Thx for your reply!
Not shure how i would check if the shapes overlap perfectly. I did check for Vertices being very close but not exacly in the same spot, but that doesn't produce meaningful results in cases where the touching corners of one shape don't lie on the corners of the other but instead lie on the connecting lines between two points of the other shape. (Not touching corners, but corners touching the side of another shape).

I did try using PGS_Conversion.roundVertexCoords() which seems to help in some cases, making the non-unision more rare, but not completely go away.

Just thought i'd share my findings - not shure what to try next... thx anyways!
